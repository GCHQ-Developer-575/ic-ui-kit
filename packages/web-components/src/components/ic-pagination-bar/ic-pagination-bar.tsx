import {
  Component,
  Prop,
  Element,
  Event,
  EventEmitter,
  State,
  Listen,
  h,
} from "@stencil/core";
import { IcThemeForeground } from "../../interface";

@Component({
  tag: "ic-pagination-bar",
  styleUrl: "ic-pagination-bar.css",
  shadow: true,
})
export class PaginationBar {
  @Element() el: HTMLIcPaginationBarElement;

  /**
   * Emitted when a page is navigated to via the 'go to' input.
   */
  @Event() icPageChange: EventEmitter<{ value: number }>;

  /**
   * Emitted when the items per page option is changed.
   */
  @Event() icItemsPerPageChange: EventEmitter<{ value: number }>;

  private setDefaultItemsPerPageOptions = () => {
    this.itemsPerPageOptions =
      this.totalItems <= 100
        ? [
            { label: "10", value: "10" },
            { label: "25", value: "25" },
            { label: "50", value: "50" },
            { label: "All", value: String(this.totalItems) },
          ]
        : [
            { label: "25", value: "25" },
            { label: "100", value: "100" },
            { label: "1000", value: "1000" },
            { label: "All", value: String(this.totalItems) },
          ];
  };

  private trimItemsPerPageOptions = () => {
    if (
      this.itemsPerPageOptions === undefined ||
      this.itemsPerPageOptions === null
    ) {
      this.itemsPerPageOptions = this.itemsPerPageOptions.slice(0, 3);
      this.itemsPerPageOptions.push({
        label: "All",
        value: String(this.totalItems),
      });
    }

    for (let i = 0; i < this.itemsPerPageOptions.length - 1; i++) {
      if (this.totalItems < Number(this.itemsPerPageOptions[i].value)) {
        this.itemsPerPageOptions.splice(
          i,
          this.itemsPerPageOptions.length - (i + 1)
        );
      }
    }
  };

  private setDefaultItemsPerPage = () => {
    this.itemsPerPage = this.itemsPerPageOptions[0].value;
  };

  private setNumberPages = () => {
    this.totalPages = Math.ceil(this.totalItems / Number(this.itemsPerPage));
  };

  private setUpperBound = () => {
    this.upperBound = Math.min(
      this.lowerBound + Number(this.itemsPerPage) - 1,
      this.totalItems
    );
  };

  private changeItemsPerPage = () => {
    const select = this.el.shadowRoot.querySelector("ic-select");
    const value = select.value;
    this.itemsPerPage = value;
    this.setNumberPages();
    this.lowerBound = 1;
    this.setUpperBound();
    const pagination = this.el.shadowRoot.querySelector("ic-pagination");
    pagination.setCurrentPage(1);
    this.currentPage = 1;
    this.icPageChange.emit({ value: 1 });
    this.icItemsPerPageChange.emit({ value: Number(this.itemsPerPage) });
  };

  private changePage = (page: number) => {
    if (page === 1) {
      this.currentPage = 1;
      this.lowerBound = 1;
      this.setUpperBound();
    } else {
      this.currentPage = page;
      this.lowerBound = (page - 1) * Number(this.itemsPerPage) + 1;
      this.setUpperBound();
    }
  };

  private goToPage = () => {
    const input = this.el.shadowRoot.querySelector("ic-text-field");
    const page = Number(input.value);
    if (page <= this.totalPages && page > 0) {
      this.changePage(page);
      const pagination = this.el.shadowRoot.querySelector("ic-pagination");
      pagination.setCurrentPage(page);
      this.currentPage = page;
      input.value = "";
      this.icPageChange.emit({ value: page });
    }
  };

  private handleKeydown = (ev: KeyboardEvent) => {
    if (ev.key === "Enter") {
      this.goToPage();
    }
  };

  private setGoToPageInputStyles = () => {
    const textField = this.el.shadowRoot?.querySelector(".go-to-page-input");
    if (textField !== undefined) {
      const input = textField?.shadowRoot?.querySelector("input");
      if (input !== undefined) {
        input.style.textAlign = "center";
        input.style.padding = "0";
      }
    }
  };

  /**
   * Total number of items to be displayed across all pages.
   */
  @Prop() totalItems!: number;

  /**
   * Whether total number of items and current item range or total number of pages and current page is displayed.
   */
  @Prop() paginationType?: "data" | "page" = "page";

  /**
   * Whether the displayed pagination is simple or complex.
   */
  @Prop() paginationControl?: "simple" | "complex" = "simple";

  /**
   * What label will be used in place of 'items' if paginationType is data, should be capitalised.
   */
  @Prop() itemLabel?: string = "Item";

  /**
   * What label will be used in place of 'Page' if paginationType is page, should be capitalised.
   */
  @Prop() pageLabel?: string = "Page";

  /**
   * What options will be displayed for 'items per page' select input, maximum of 4 options including a required 'All' option with value equal to total number of items.
   */
  @Prop({ mutable: true }) itemsPerPageOptions?: {
    label: string;
    value: string;
  }[];

  /**
   * Whether the select input to control 'items per page' should be displayed.
   */
  @Prop() showItemsPerPageControl?: boolean = false;

  /**
   * Whether the number of total items and current item range or number of total pages and current page should be displayed.
   */
  @Prop() showItemsPerPage?: boolean = true;

  /**
   * Whether the 'go to page' control should be displayed.
   */
  @Prop() showGoToPageControl?: boolean = false;

  /**
   * Sets the alignment of the items in the pagination bar.
   */
  @Prop() alignment?: "left" | "right" | "space-between" = "right";

  /**
   * Sets the styling for the items in the pagination bar.
   */
  @Prop() appearance?: IcThemeForeground = "default";

  @Listen("icPageChange")
  pageChangeHandler(ev: CustomEvent) {
    const page = ev.detail.value;
    this.changePage(page);
  }

  @State() itemsPerPage: string;

  @State() totalPages: number;

  @State() currentPage: number = 1;

  @State() lowerBound: number = 1;

  @State() upperBound: number;

  componentWillLoad(): void {
    if (
      this.itemsPerPageOptions === undefined ||
      this.itemsPerPageOptions === null
    ) {
      this.setDefaultItemsPerPageOptions();
    }
    this.trimItemsPerPageOptions();
    this.setDefaultItemsPerPage();
    this.setNumberPages();
    this.setUpperBound();
  }

  componentDidLoad(): void {
    this.setGoToPageInputStyles();
  }

  render() {
    const {
      appearance,
      alignment,
      itemsPerPageOptions,
      paginationControl,
      paginationType,
      showItemsPerPage,
      showItemsPerPageControl,
      showGoToPageControl,
    } = this;

    return (
      <div
        class={{
          ["pagination-bar"]: true,
          [`pagination-bar-${alignment}`]: true,
        }}
      >
        {showItemsPerPage || showItemsPerPageControl ? (
          <div class="item-controls">
            {showItemsPerPageControl ? (
              <div class="items-per-page-holder">
                <ic-typography
                  class={{
                    [`pagination-text-${appearance}`]: true,
                    ["items-per-page-control-label"]: true,
                  }}
                  variant="label"
                >
                  {this.itemLabel}s per {this.pageLabel.toLowerCase()}
                </ic-typography>
                <ic-select
                  small
                  label="items-per-page-input"
                  class="items-per-page-input"
                  hideLabel
                  options={itemsPerPageOptions}
                  value={this.itemsPerPage}
                  onIcChange={() => this.changeItemsPerPage()}
                ></ic-select>
              </div>
            ) : null}
            {showItemsPerPage && paginationType === "data" ? (
              <ic-typography
                class={{
                  [`pagination-text-${appearance}`]: true,
                  ["item-pagination-label"]: true,
                }}
                variant="label"
              >
                {this.lowerBound} - {this.upperBound} of {this.totalItems}{" "}
                {this.itemLabel.toLowerCase()}s
              </ic-typography>
            ) : showItemsPerPage ? (
              <ic-typography
                class={{
                  [`pagination-text-${appearance}`]: true,
                  ["page-pagination-label"]: true,
                }}
                variant="label"
              >
                {this.pageLabel} {this.currentPage} of {this.totalPages}
              </ic-typography>
            ) : null}
          </div>
        ) : null}
        <div class="pagination-controls">
          <div class="pagination-holder">
            <ic-pagination
              appearance={appearance}
              type={paginationControl}
              pages={this.totalPages}
            ></ic-pagination>
          </div>
          {showGoToPageControl ? (
            <div class="go-to-page-holder">
              <ic-typography
                class={{ [`pagination-text-${appearance}`]: true }}
                variant="label"
              >
                Go to {this.pageLabel.toLowerCase()}
              </ic-typography>
              <ic-text-field
                type="number"
                small
                label="go-to-page-input"
                class="go-to-page-input"
                hideLabel
                onKeyDown={(ev) => this.handleKeydown(ev)}
              ></ic-text-field>
              <ic-button
                appearance={appearance}
                variant="secondary"
                onClick={() => this.goToPage()}
                size="small"
              >
                Go
              </ic-button>
            </div>
          ) : null}
        </div>
      </div>
    );
  }
}
