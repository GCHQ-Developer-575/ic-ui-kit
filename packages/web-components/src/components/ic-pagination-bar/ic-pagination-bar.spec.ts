import { PaginationBar } from "./ic-pagination-bar";
import { Pagination } from "../ic-pagination/ic-pagination";
import { Button } from "../ic-button/ic-button";
import { newSpecPage } from "@stencil/core/testing";

describe("ic-pagination-bar", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with items per page controls", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with go to page controls", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with data pagination type", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" pagination-type="data"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with complex pagination controls", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" pagination-controls="complex"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a custom page label", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" page-label="Sheets"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a custom item label", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" pagination-type="data" item-label="Rows"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with left alignment", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" alignment="left"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with space between alignment", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" alignment="space-between"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with light appearance", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" appearance="light"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with dark appearance", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" apperance="dark"></ic-pagination-bar>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with custom items per page options", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    paginationBar.itemsPerPageOptions = [
      { label: "15", value: "15" },
      { label: "30", value: "30" },
      { label: "60", value: "60" },
      { label: "All", value: "100" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should only allow a maximum of 4 custom items per page options", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="150" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    paginationBar.itemsPerPageOptions = [
      { label: "25", value: "25" },
      { label: "50", value: "50" },
      { label: "75", value: "75" },
      { label: "100", value: "100" },
      { label: "All", value: "150" },
    ];

    page.rootInstance.trimItemsPerPageOptions();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should remove items per page options larger than the maximum number of items", async () => {
    const page = await newSpecPage({
      components: [PaginationBar],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    paginationBar.itemsPerPageOptions = [
      { label: "25", value: "25" },
      { label: "50", value: "50" },
      { label: "150", value: "150" },
      { label: "All", value: "100" },
    ];

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  it("should adjust the total page count when items per page is changed", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination],
      html: `<ic-pagination-bar total-items="100" show-items-per-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const select = paginationBar.shadowRoot.querySelector("ic-select");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    expect(text.textContent).toEqual("Page 1 of 10");

    select.value = "25";

    page.rootInstance.changeItemsPerPage();

    page.rootInstance.setNumberPages();

    await page.waitForChanges();

    expect(select.value).toEqual("25");

    expect(text.textContent).toEqual("Page 1 of 4");
  });

  it("should change page when the pagination controls are clicked", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination, Button],
      html: `<ic-pagination-bar total-items="100"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const pagination = paginationBar.shadowRoot.querySelector("ic-pagination");

    const nextButton = pagination.shadowRoot.querySelector(
      "#next-page-button"
    ) as HTMLElement;

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    expect(text.textContent).toEqual("Page 1 of 10");

    await nextButton.click();

    page.rootInstance.changePage(2);

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 2 of 10");
  });

  it("should change page when the go to page button is clicked", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const button = paginationBar.shadowRoot.querySelector("ic-button");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    const currentPage = page.rootInstance.currentPage;

    expect(text.textContent).toEqual("Page 1 of 10");

    expect(currentPage).toEqual(1);

    input.value = "3";

    button.click();

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 3 of 10");
  });

  it("should change page when enter is pressed on the go to page input", async () => {
    const page = await newSpecPage({
      components: [PaginationBar, Pagination],
      html: `<ic-pagination-bar total-items="100" show-go-to-page-control="true"></ic-pagination-bar>`,
    });

    const paginationBar = document.querySelector("ic-pagination-bar");

    const input = paginationBar.shadowRoot.querySelector("ic-text-field");

    const text = paginationBar.shadowRoot.querySelector(
      ".page-pagination-label"
    );

    const currentPage = page.rootInstance.currentPage;

    const event = new KeyboardEvent("keydown", { key: "Enter" });

    expect(text.textContent).toEqual("Page 1 of 10");

    expect(currentPage).toEqual(1);

    input.value = "3";

    input.focus();

    page.rootInstance.handleKeydown(event);

    await page.waitForChanges();

    expect(text.textContent).toEqual("Page 3 of 10");
  });
});
