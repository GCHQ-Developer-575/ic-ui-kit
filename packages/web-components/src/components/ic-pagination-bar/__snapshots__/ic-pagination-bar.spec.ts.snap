// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-pagination-bar should only allow a maximum of 4 custom items per page options 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="150">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="25"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 6
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="6" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should remove items per page options larger than the maximum number of items 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="10"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render 1`] = `
<ic-pagination-bar total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with a custom item label 1`] = `
<ic-pagination-bar item-label="Rows" pagination-type="data" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="item-pagination-label pagination-text-default" variant="label">
          1 - 10 of 100 rowss
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with a custom page label 1`] = `
<ic-pagination-bar page-label="Sheets" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Sheets 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with complex pagination controls 1`] = `
<ic-pagination-bar pagination-controls="complex" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with custom items per page options 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="10"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with dark appearance 1`] = `
<ic-pagination-bar apperance="dark" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with data pagination type 1`] = `
<ic-pagination-bar pagination-type="data" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="item-pagination-label pagination-text-default" variant="label">
          1 - 10 of 100 items
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with go to page controls 1`] = `
<ic-pagination-bar show-go-to-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
        <div class="go-to-page-holder">
          <ic-typography class="pagination-text-default" variant="label">
            Go to page
          </ic-typography>
          <ic-text-field class="go-to-page-input" hidelabel="" label="go-to-page-input" small="" type="number"></ic-text-field>
          <ic-button appearance="default" size="small" variant="secondary">
            Go
          </ic-button>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with items per page controls 1`] = `
<ic-pagination-bar show-items-per-page-control="true" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <div class="items-per-page-holder">
          <ic-typography class="items-per-page-control-label pagination-text-default" variant="label">
            Items per page
          </ic-typography>
          <ic-select class="items-per-page-input" hidelabel="" label="items-per-page-input" small="" value="10"></ic-select>
        </div>
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with left alignment 1`] = `
<ic-pagination-bar alignment="left" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-left">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with light appearance 1`] = `
<ic-pagination-bar appearance="light" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-right">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-light" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="light" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;

exports[`ic-pagination-bar should render with space between alignment 1`] = `
<ic-pagination-bar alignment="space-between" total-items="100">
  <mock:shadow-root>
    <div class="pagination-bar pagination-bar-space-between">
      <div class="item-controls">
        <ic-typography class="page-pagination-label pagination-text-default" variant="label">
          Page 1 of 10
        </ic-typography>
      </div>
      <div class="pagination-controls">
        <div class="pagination-holder">
          <ic-pagination appearance="default" pages="10" type="simple"></ic-pagination>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</ic-pagination-bar>
`;
